src/basic/ARRAY_OF_PTRS-Sycl.cpp

```diff
@@ -0,0 +1,84 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2017-24, Lawrence Livermore National Security, LLC
+// and RAJA Performance Suite project contributors.
+// See the RAJAPerf/LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include "ARRAY_OF_PTRS.hpp"
+
+#include "RAJA/RAJA.hpp"
+
+#if defined(RAJA_ENABLE_SYCL)
+
+#include "common/SyclDataUtils.hpp"
+
+#include <iostream>
+
+namespace rajaperf
+{
+namespace basic
+{
+
+
+template < size_t work_group_size >
+void ARRAY_OF_PTRS::runSyclVariantImpl(VariantID vid)
+{
+  const Index_type run_reps = getRunReps();
+  const Index_type ibegin = 0;
+  const Index_type iend = getActualProblemSize();
+
+  auto res{getSyclResource()};
+  auto qu = res.get_queue();
+
+  ARRAY_OF_PTRS_DATA_SETUP;
+
+  if ( vid == Base_SYCL ) {
+
+    ARRAY_OF_PTRS_Array x_array = x;
```

Comment:
Why are we doing this here?

Reply:
I don't understand the question. Isn't this following the same implementation pattern as the CUDA version?

Reply:
I think question is why doesn't the RAJA variant require it?

Reply:
I remember now, when you try to pass an array as a function argument it just passes a pointer, so for cuda and hip where we have a global function we wrap the array in a struct to pass it by value. Here where it is captured in a lambda that is not necessary.